_rules/SC1-3-1-aria-required-owned-element.md

```diff
@@ -0,0 +1,150 @@
+---
+name: ARIA required owned elements
+
+description: |
+	The rule checks existence of role(s) on required owned elements.
+
+success_criterion:
+- 1.3.1
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Audrey Maniez
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any element that is [exposed to assistive technologies](#exposed-to-assistive-technologies) with a [non empty](#non-empty) `role` attribute whose value is applicable to required owned elements.
+
+**Note:** I
+
+### Expectation
+
+Each target element has atleast one owned element with a relevant `role` attribute.
+
+## Assumptions
+
+*There are currently no assumptions/ add any assumptions as necessary*
+
+## Accessibility Support
+
+*There are no major accessibility support issues known for this rule/ or add any support details as necessary*
+
+## Background
+
+- [Required Owned Element](https://www.w3.org/TR/wai-aria-1.1/#mustContain)
+
+## Test Cases
+
+### Passed
+
+#### Pass example 1
+
+Owned `ul` element with role `list` has `span` element with role `listitem`.
+
+```html
+<ul role='list'>
+	<span role='listitem'>
+	</span>
+	<span>
+		<!-- atleast one element is expected have a role, not mandatory on all children -->
+	</span>
+</ul>
+```
+
+#### Pass example 2
+
+Owned `ol` element with role `tablist` has `li` element with role `tab`.
+
+```html
+<ol role='tablist'>
+	<li role='tab'>
+	</li>
+</ol>
+```
+
+#### Pass example 3
+
+Owned `ul` element with role `list` has `li` element which has an implcit semantic role.
+
+```html
+<ul role='list'>
+	<li></li> <!-- implicit role -->
+</ul>
+```
+
+#### Pass example 4
+
+Multiple owned elements with relevant nested roles.
+
+```html
+<table role='grid'>
+	<tr role='row'>
+		<span role='cell'>
+		</span>
+	</tr>
+</table>
+```
+
+#### Pass example 5
+
+Multiple satisfying roles on owned element(s).
+
+```html
+<ul role='menu'>
+	<li></li> <!-- implicit role -->
+	<li role='menuitem'></li>
+	<li role='menuitemradio'></li>
+</ul>
+```
+
```

Comment:
Suggest adding a test case that highlights that the required owned element does not have to be a child of the role, but could be any descendant, e.g.:

```html
<div role='menu'>
	<div>
		<div>
			<div>Some content</div>
		</div>
	</div>
	<div>
		<div>
			<div role=´menuitem´></div>
		</div>
	</div>
</div>
```

Reply:
I have added a test case for above and a note to the accessibility support section.

Reply:
@audreymaniez is correct. You have to explicitly set `role=presentation` on an element if you want to get not just the children, but any descendants to be considered owned elements. I haven't found any description of this in the ARIA specs, but if you look at the ARIA best practices you'll see this used consistently.

@ShadowBB The reason I believe this is required by WCAG is that without a correct parent / child relationship a number of things can go wrong:

- For combobox, you need the owned elements. Without a textbox and some sort of list to pull suggestions from, that element isn't a combobox. Those owned elements are required because without them it wouldn't be a combobox.
- Everything else is about computing group size, the size of the group can't be correctly expressed to AT if you can't programmatically determine where the group starts and where it ends. It's like instead of having a single list of 5 items, you'd get 5 lists with 1 item each. That kind of stuff is a violation of WCAG.

Can you give an example of when you think this isn't a WCAG 2 conformance issue? If anyone's interested, we could maybe split this rule up into one rule for combobox, and a second for group sizes? Personally I don't see too much value in that, but I suppose it identifies the issue a little more explicitly.

Reply:
> I haven't found any description of this in the ARIA specs, but if you look at the ARIA best practices you'll see this used consistently.

@WilcoFiers I just had a look at the ARIA Authoring Practices and the one thing they do consistently mention is that when relations are involved, a given child "is either a DOM descendant of or owned by" its parent. The only place where e.g. `role=none` is used is when removing the default semantics of elements that are put between a parent and its owned descendant. All of this fits perfectly with the definition of "owned element" in ARIA:

> An 'owned element' is any DOM descendant of the element, any element specified as a child via aria-owns, or any DOM descendant of the owned child.

If anything, I see this as an accessibility support question as e.g. VoiceOver supports all of this just fine.

Reply:
I believe these comments are on a different tangent as to what the rule addresses. So will consider this resolved.

Reply:
@jkodu I don't agree that this is resolved, or that it's not relevant. The following test case is completely broken and yet it seems to pass this rule:

```html
<div role=Listbox>
  <div role=alert>
    <div role=listitem>Item 1</div>
    <div role=listitem>Item 2</div>
  </div>
</div>
```

I'm in the process of reverse engineering how browsers actually work out owned items, and if it even matters. It's a bit more complicated than it may seem on the surface.

Reply:
@WilcoFiers There's a ["owned by" definition](https://github.com/auto-wcag/auto-wcag/pull/255/files#diff-e8efc5e044aa522fbb04b4f072890fa3) in writing for the #255. Is the definition enough to clarify that kind of example you give is a fail one ? 

Actual expectation:
> For each test target, at least one instance of one [WAI-ARIA required owned element](https://www.w3.org/TR/wai-aria-1.1/#mustContain) is present.

Could be rephrase to add the owned by definition link:

> The target element [owned](#owned-by) at least one instance of one of its [WAI-ARIA required owned element](https://www.w3.org/TR/wai-aria-1.1/#mustContain).

or do you think are more details are needed in the expectation section ? 

And we should add a fail example that demonstrate that (not sure of the description of the example though): 

> #### Failed Example x
> 
> Element with role `list` is not the closest ancestor of element with role `listitem`, there is a role `alert` in between that  break the relationship. The role `list` is considered having no WAI-ARIA required owned element.
> 
````html
<div role="list">
  <div role="alert">
    <div role="listitem">Item 1</div>
    <div role="listitem">Item 2</div>
  </div>
</div>